### PR TITLE
Fix startup issue linux

### PR DESCRIPTION
**Pull request by a1rwulf**
## Description
Fixes the startup crashes that occur since:
https://github.com/xbmc/xbmc/pull/17458

@MilhouseVH can you please also test this?

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by MilhouseVH** [comment]
@a1rwulf thanks! Looks good to me - over a dozen starts and zero crashes.

**Comment by a1rwulf** [comment]
@MilhouseVH ok cool - imho good to merge when CI is green and I didn't fuck up the code formatting :)

**Comment by fritsch** [comment]
Great! Wondering what Coverity says. That's its job :-)

**Review by fritsch — APPROVED** [review]

File: xbmc/network/Socket.cpp [hunk 1]
```diff
@@ -81,7 +82,8 @@ bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
           m_iSock = INVALID_SOCKET;
         }
 
-        closesocket(testSocket);
+        if (testSocket != INVALID_SOCKET)
```
Comment (yol) [inline xbmc/network/Socket.cpp:85]: this code will never run

**Comment by yol** [comment]
This is really crying for a scope guard :o)

File: xbmc/network/Socket.cpp (hunk 1 above)
Comment (a1rwulf) [reply xbmc/network/Socket.cpp:85]: How comes?
This code is definitely run otherwise it wouldn't fix anything.
I saw it in the debugger on my system!

Comment (yol) [reply xbmc/network/Socket.cpp:85]: I confused `testSocket` and `m_iSock` - disregard :-)

Comment (yol) [reply xbmc/network/Socket.cpp:85]: lulz I was still right xD

**Pull request merged**